pkg/services/services.go

```diff
@@ -0,0 +1,65 @@
+package services
+
+import (
+	"sync"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/kubecost/cost-model/pkg/log"
+)
+
+// HTTPService defines an implementation prototype for an object capable of registering
+// endpoints on an http router which provide an service relevant to the cost-model.
+type HTTPService interface {
+	// Register assigns the endpoints and returns an error on failure.
+	Register(*httprouter.Router) error
+}
+
+// HTTPServices defines an implementation prototype for an object capable of managing and registering
+// predefined HTTPService routes
+type HTTPServices interface {
+	// Add a HTTPService implementation for registration
+	Add(service HTTPService)
+
+	// RegisterAll registers all the services added with the provided router
+	RegisterAll(*httprouter.Router) error
+}
+
+type defaultHTTPServices struct {
+	sync.Mutex
+	services []HTTPService
+}
+
+// Add a HTTPService implementation for
+func (dhs *defaultHTTPServices) Add(service HTTPService) {
+	if service == nil {
+		log.Warningf("Attempting to Add nil HTTPService")
+		return
+	}
+
+	dhs.Lock()
+	defer dhs.Unlock()
+
+	dhs.services = append(dhs.services, service)
+}
+
+// RegisterAll registers all the services added with the provided router
+func (dhs *defaultHTTPServices) RegisterAll(router *httprouter.Router) error {
+	dhs.Lock()
+	defer dhs.Unlock()
+
+	for _, svc := range dhs.services {
+		svc.Register(router)
+	}
+
+	return nil
+}
+
+// NewCostModelServices creates an HTTPServices implementation containing any predefined
+// http services used with the cost-model
+func NewCostModelServices() HTTPServices {
+	return &defaultHTTPServices{
+		services: []HTTPService{
+			NewClusterManagerService(),
+		},
+	}
+}
```

Comment:
Is there a good way to document this new pattern? Something like "all new services should be `HTTPService`s and they should be added to the default set of services created by `NewCostModelServices`". I know we don't really have good dev documentation right now, but this is something that could go in a `CONTRIBUTING.md` or `DEVELOPMENT.md`.

Reply:
I'm going to defer immediate documentation for this specific design for a few reasons:

1. I don't have any expectations (currently) of enforcing this pattern on any existing http handlers delegating an internal API. The reason why is that the one service I added was an API that stuck out as odd in the router and in the pkg. It appears to be a modifiable resource (which it was designed that way initially), but it was written down to just configmap proxy (sadly). It's a unique sort of self-contained API that will likely be used in our stack later.
2. This particular API is something I am imagining being something on our hosted solution (real REST implementation with actual stateful resources). 

So the reasoning behind the shifting around here is to remove the black sheep API from front and center, and to prototype a general RESTful http service springboard for the future. Hope that all makes sense :)